sprint-mission-7/queries.sql

```diff
@@ -0,0 +1,169 @@
+/*
+  다음 경우들에 대해 총 14개의 SQL 쿼리를 작성해 주세요.
+  예시로 값이 필요한 경우 적당한 값으로 채워넣어서 작성하면 됩니다. 
+*/
+
+/*
+  1. 내 정보 업데이트 하기
+  - 닉네임을 "test"로 업데이트
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+
+update users
+set nickname = 'test', updated_at = now()
+where id = 1;
+
+/*
+  2. 내가 생성한 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+
+select *
+from products
+where user_id = 1
+order by created_at desc
+limit 10 offset (3 - 1) * 10;
+
+/*
+  3. 내가 생성한 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+--count(user_id)로 작성해도 스키마 상 not null이기 때문에 값은 같지만 
+--나중에 컬럼이 nullable로 바뀌거나할 경우 join으로 컬럼이 null이 섞여도 의미가 흔들리지 않아서 * 권장
+
+select count(*)  
+from products
+where user_id = 1;
+
+/*
+  4. 내가 좋아요 누른 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+
+select products.name as "like_product_name",
+products.description,
+products.price,
+products.tags,
+products.images,
+products.created_at,
+likes.created_at as "like_created_at"
+from products join likes
+on products.id = likes.product_id
+where likes.user_id = 1 and likes.product_id is not null
+order by likes.created_at desc
+limit 10 offset (3 -1) * 10;
+
+/*
+  5. 내가 좋아요 누른 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+
+select user_id, count(*) as total_count
+from likes
+where user_id = 1
+group by user_id;
```

Comment:
이 쿼리는 특정 유저(id=1)의 좋아요 개수를 조회하는데, `where user_id = 1` 조건 때문에 결과가 항상 1개의 행만 반환됩니다. 
따라서 `group by user_id`와 `select user_id`는 불필요합니다.

더 간결하게 한다면 이렇게 작성할 수 있어요.
```suggestion
/*
  5. 내가 좋아요 누른 상품의 총 개수
  - 현재 로그인한 유저 id가 1이라고 가정
*/

SELECT count(*) as total_count
FROM likes
WHERE user_id = 1 and product_id is not null;
```
여러 유저의 좋아요 개수를 한 번에 조회한다면 `GROUP BY`가 필요하지만, 특정 유저 1명만 조회할 때는 필요 없습니다. 
또한 통합 `likes` 테이블을 사용하고 계시니 `product_id is not null` 조건을 추가하여 상품 좋아요만 카운트하는 것이 정확합니다.